packages/components/src/internal/components/samples/SamplesEditableGrid.spec.tsx

```diff
@@ -0,0 +1,173 @@
+import { List, fromJS } from 'immutable';
+
+import { QueryGridModel } from '../../QueryGridModel';
+import { QueryInfo } from '../../../public/QueryInfo';
+import { makeTestQueryModel } from '../../../public/QueryModel/testUtils';
+import { SchemaQuery } from '../../../public/SchemaQuery';
+import { DataClassDataType, SampleTypeDataType } from '../entities/constants';
+import { EntityChoice, IEntityTypeOption } from '../entities/models';
+
+import { getLineageEditorUpdateColumns, getUpdatedLineageRows } from './SamplesEditableGrid';
+
+let DATA = fromJS({
+    '1': {
+        RowId: 1,
+        'MaterialInputs/One': List(),
+        'MaterialInputs/Two': List(),
+    },
+    '2': {
+        RowId: 2,
+        'MaterialInputs/One': List(),
+        'MaterialInputs/Two': List(),
+    },
+});
+DATA = DATA.setIn(
+    ['1', 'MaterialInputs/One'],
+    List.of({
+        value: 1,
+        displayValue: 'A',
+    })
+);
+DATA = DATA.setIn(
+    ['1', 'MaterialInputs/Two'],
+    List.of(
+        {
+            value: 2,
+            displayValue: 'B',
+        },
+        {
+            value: 3,
+            displayValue: 'C',
+        }
+    )
+);
+const ORIGINAL_MODEL = new QueryGridModel({ data: DATA });
+
+describe('getUpdatedLineageRows', () => {
+    test('no changes', () => {
+        const updatedRows = getUpdatedLineageRows(
+            [
+                { RowId: 1, 'MaterialInputs/One': 'A', 'MaterialInputs/Two': 'B, C' },
+                { RowId: 2, 'MaterialInputs/One': '', 'MaterialInputs/Two': '' },
+            ],
+            ORIGINAL_MODEL,
+            []
+        );
+        expect(updatedRows.length).toBe(0);
+    });
+
+    test('add to existing parent', () => {
+        const updatedRows = getUpdatedLineageRows(
+            [
+                { RowId: 1, 'MaterialInputs/One': 'A', 'MaterialInputs/Two': 'B, C, D' },
+                { RowId: 2, 'MaterialInputs/One': '', 'MaterialInputs/Two': '' },
+            ],
+            ORIGINAL_MODEL,
+            []
+        );
+        expect(updatedRows.length).toBe(1);
+        expect(updatedRows[0].RowId).toBe(1);
```

Comment:
Is there any value in checking the other fields here (validating that MaterialInputs/Two has value "B, C, D")?

Reply:
`getUpdatedLineageRows` should just be filtering the `updatedRows` array and not changing its values in any way...but never hurts to check. Added.